ruby/lib/gherkin3/token_scanner.rb

```diff
@@ -11,15 +13,23 @@ module Gherkin3
   # to look for Gherkin keywords for the associated language. The keywords are defined 
   # in gherkin-languages.json.
   class TokenScanner
+    protected
+
+    attr_reader :io
```

Comment:
Ok. I can do this, thought that should be better part of the token scanner and not any business of the parser.

You already have a "if"-statement here https://github.com/dg-ratiodata/gherkin3/commit/d9085a51cc39f516795f2ef2b0b5f6bdcb415a81#diff-a5a70ea60d3411ec934a6b3412ff8af6L17 I see no good reason why to introduce another one in the parser.